pkg/client/msp/ca.go

```diff
@@ -195,3 +197,19 @@ type GetCAInfoResponse struct {
 	// Version of the server
 	Version string
 }
+
+// CSRInfo is Certificate Signing Request (CSR) Information
+type CSRInfo struct {
+	CN           string
+	Names        []csr.Name
+	Hosts        []string
+	KeyRequest   *BasicKeyRequest
+	CA           *csr.CAConfig
```

Comment:
This API shouldn't have dependencies on "github.com/cloudflare/cfssl/csr". CAConfig should be created here, and any mapping between this API and third party should be hidden behind the API. While we have a freedom to design our API structures independently from the third party, it is useful to keep them close, as we keep extending support for Fabric CA feature

Reply:
I strongly disagree, to create a new struct and re-map it as done in CSR struct seems a bad thing to me, CA depends on cloudflare structure, this dependency is already packaged with sdk code, btw. 
I don't see how creating same structs and "convert this type to that type" functions, where those types are the same
what do you think

Reply:
Neither fabric-ca nor cloudflare/cfssl are used directly in the exposed API, all uses are internal. I see that we do use cloudflare/cfssl in a couple of tests, perhaps we shouldn't, we can easily change that. Usage of cloudflare/cfssl internally is an implementation detail, shouldn't be in public API.

You wanted to add only hosts to CSR info, and my thinking was, if we keep adding the missing pieces of Fabric CA support, we will likely end up with structures similar to what the CA is using. This is why I thought we should introduce CSRInfo, to group hosts and future (currently missing) elements together in a way it makes sense. CSRInfo could initially have only hosts, and whatever you currently need. Ideally, all additions should have proper integration tests.